src/firebolt/async_db/cursor.py

```diff
@@ -584,6 +591,49 @@ def setinputsizes(self, sizes: List[int]) -> None:
     def setoutputsize(self, size: int, column: Optional[int] = None) -> None:
         """Set a column buffer size for fetches of large columns (does nothing)."""
 
+    @check_not_closed
+    async def cancel(self, query_id: str) -> None:
+        """Cancel a server-side async query."""
+        try:
+            resp = await self._api_request(
+                parameters={"query_id": query_id}, url_suffix="cancel"
+            )
```

Comment:
add `self._raise_if_error`. Also check if all it's functionality fits here

Reply:
I think you're right. I removed this check. Instead, after a cancel the user can check the status of the query.